### DISCUSSION TITLE
Better support for offline environments.

### DISCUSSION THREAD

**Discussion by gaby** [dani-garcia/vaultwarden#2361] [Ideas]
I have been using Vaultwarden in an offline/airgap environment and have noticed a few things that would make the experience better for the end-user.

* When vaultwarden sends an email is uses images hosted on the internet, so emails in offline environments look broken.
* It would be very beneficial if there was a way to set a `Color` for an organization and make the UI menu change to the color of the organization. This would help with users that are part of multiple organization with credentials for the same services. 
* Most organizations run custom-ca's, in the current state not straightforward how to add custom CA's for services like SMTP. Better documentation on this is needed in the `Wiki` section.

**Comment by BlackDex** [comment]
Well, this is kinda a niche case here i think. And i think most of these can be solved by creating an init script as mentioned here: https://github.com/dani-garcia/vaultwarden/wiki/Starting-a-Container#customizing-container-startup

Regarding the images of the emails, you can try to create base64 strings of those images and embed them. This does mean that you need to override the templates, which is not safe when updating and those templates have changed. But embedding those images by code also causes issues. Some mail servers or clients mark those kind of mails as spam, or do not even allow them to be sent/received.

Though it could be added maybe as a feature-flag, but that would take some effort to add that feature.

The other two should be able to be fixable via an init script.
```bash
#!/usr/bin/env sh

# Change navbar color
echo -e '\n.navbar { background-color: #FF0000 !important; }\n' >> /web-vault/app/main*.css
```

For the CA.
Add all the custom CA's you have into a folder and mount this via a volume like so:
`-v /local/path/to/ca/files/:/usr/local/share/ca-certificates/custom/`

Make sure all the CA's in that folder have an extension of `.crt` else it will not work.

Then via an init script:
```bash
#!/usr/bin/env sh

update-ca-certificates
```

And, regarding the wiki, you are free to update or add items.

**Comment by gaby** [comment]
@BlackDex That trick for changing the navigation color works wonders, but what I'm interested in is having a  `Different Color` per organization. 

I was able to solve the custom CA problem using an init script, as suggested.

**Reply by BlackDex** [reply]
Per org should be doable with css and some long checks but not easy to implement everytime.

For a feature like that i suggest to report that upstream if it is not already there.

**Reply by gaby** [reply]
Will do, thanks!
